src/preact.d.ts

```diff
@@ -52,7 +52,8 @@ declare namespace preact {
 		key?: Key | null;
 	}
 
-	type RenderableProps<P> = Readonly<P> & Readonly<{ children?: ComponentChildren }>;
+	type RenderableProps<P> = Readonly<P & Attributes & { children?: ComponentChildren }>;
+	type ClassRenderableProps<P> = RenderableProps<P> & Readonly<ClassAttributes<any>>;
```

Comment:
Maybe it makes more sense to have the ref defined right next to the `children` at the `RenderableProps`. That way we do not need the ClassRenderableProps and we have ref working for the SFC. What do you think about that?

Reply:
@andrewiggins what are your thoughts? Apart from this the pr is ready to be merged 🎉👍

Reply:
@marvinhagemeister whoops! Thought I had pushed an update for this, but I forgot to. I've done what @valotas suggested and added a line in the test file to validate that refs work for SFCs. Thanks for following up.

P.S. Congrats on the 8.2.8 release! Looking forward to integrating it into my project :)